nexus/src/app/disk.rs

```diff
@@ -510,8 +510,11 @@ impl super::Nexus {
             .project_delete_snapshot(opctx, &authz_snapshot, &db_snapshot)
             .await?;
 
-        // Kick off volume deletion saga
+        // Kick off volume deletion saga(s)
         self.volume_delete(db_snapshot.volume_id).await?;
```

Comment:
Just my ignorance: is this volume id the origin of the snapshot?  Why do we soft-delete that when we delete the snapshot?

Reply:
If a disk was created out of a snapshot, then the snapshot is deleted, we want to preserve the parts of the disk's volume so that the disk's blocks are still valid. We have to soft delete when resources are shared like this.